C/services/core/core_management_api.cpp

```diff
@@ -63,12 +139,58 @@ void CoreManagementApi::defaultResource(shared_ptr<HttpServer::Response> respons
 /**
  * Construct a microservices management API manager class
  */
-CoreManagementApi::CoreManagementApi(const string& name, const unsigned short port) : ManagementApi(name, port)
+CoreManagementApi::CoreManagementApi(const string& name,
+				     const unsigned short port) : ManagementApi(name, port)
 {
+
+	unsigned short storageLayerRetries = 10;
+
+	// Instantiate ConfigurationManager via Storage Layer
+	ServiceRegistry* registry = ServiceRegistry::getInstance();
+	if (!registry)
+	{
+		// TODO: add exception
+		throw;
+	}
+
+	// Wait for storage layer registration
+	ServiceRecord* storageDetails = 0;
+	while (storageLayerRetries)
```

Comment:
I do not think this is a good idea. The core management API needs to be up and running for the storage layer to register. Before it can register it must be started and passed the port of the core management API. For this to happen the constructor must have returned so that we have access to a fully constructed class. This functionality needs to be outside of the constructor.

Reply:
yes, code changed: the registerService now triggers ConfigurationManager instance and URL entry points.
Next commit will add the new functionality.

The storage service registration could set a callback for ConfigurationManager operations: that can be added with future commits.